### PR TITLE
implementing BLAKE3 opcode

### PR DESCRIPTION
**Pull request by fengjian**

### CODE REVIEW

**Comment by fengjian** [comment]
Rebase merge all commits

File: src/isa/instr.rs [hunk 1]
```diff
@@ -883,6 +883,16 @@ pub enum DigestOp {
         /** Index of string register */ RegS,
         /** Index of `r512` register to save result to */ Reg16,
     ),
+
+    /// Computes BLAKE3 hash value
+    ///
+    /// Sets `st0` to `false` and destination register to `None` if the source register does not
+    /// contain a value
+    #[display("Blake3    {0},r256{1}")]
```
Comment (dr-orlovsky) [inline src/isa/instr.rs:891]: can you please change mnemonic so it looks more in line with the rest
```suggestion
    #[display("blake3  {0},r256{1}")]
```

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
one small change, the rest LGTM

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/AluVM/rust-aluvm/commit/bf50a8ca7fe48db13d68eb73e686e8a4a080927b

**Pull request merged**